### PR TITLE
For #1539: Add bookmark multi-select features

### PR DESCRIPTION
**Pull request by colintheshots**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -186,6 +187,13 @@ class HomeFragment : Fragment(), CoroutineScope {
         sessionObserver?.onSessionsRestored()
     }
 
+    override fun onStop() {
+        sessionObserver?.let {
+            requireComponents.core.sessionManager.unregister(it)
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: This is a crash fix required to support opening multiple private tabs at once.

File: app/src/test/java/org/mozilla/fenix/library/bookmarks/BookmarkComponentTest.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package org.mozilla.fenix.library.bookmarks
```
Comment (boek) [inline app/src/test/java/org/mozilla/fenix/library/bookmarks/BookmarkComponentTest.kt]: nit: license

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -305,6 +305,8 @@
     <string name="bookmark_menu_open_in_private_tab_button">Open in private tab</string>
     <!-- Bookmark overflow menu delete button -->
     <string name="bookmark_menu_delete_button">Delete</string>
+    <!-- Bookmark multi select title in app bar -->
```
Comment (boek) [inline app/src/main/res/values/strings.xml]: This needs a description of the string parameter :+1:

**Comment by colintheshots** [comment]
Don't merge yet. I discovered that the themes on the actionBar aren't changing correctly yet.

**Pull request merged**